test/Rings/orderings-test.jl

```diff
@@ -47,15 +47,30 @@
    M = [ 1 1 1; 1 0 0; 0 1 0 ]
    @test collect(monomials(f, M)) == collect(monomials(f, :deglex))
 
-   @test is_global(lex([x, y]))
+   @test !is_global(lex([x, y]))
    @test !is_local(lex([x, y]))
    @test !is_mixed(lex([x, y]))
    @test !is_global(neglex([y, z]))
-   @test is_local(neglex([y, z]))
+   @test !is_local(neglex([y, z]))
```

Comment:
It was wrong according to the definition in the doc string.

Reply:
I added those `is_global/local/mixed` methods and wrote the doc-strings. By "for each variable $x$ in the ring `M.R` for which `M` is defined" I meant that the property is only checked for the variables the ordering actually knows about. So "for which `M` is defined" is supposed to be about the variable not the ring. In that sense the old implementation and tests should be correct (`neglex([y, z])` is local for the variables y and z, it just doesn't know anything about x). I'm not saying that this is the "right" definition...

Reply:
Since the docstring did not change, your statements are not consistent. Or is the question what is meant by
> each variable $x$ in the ring `M.R` for which `M` is defined..

Is the ordering here defined for `x` or not?

Reply:
Well, there is `.vars` for an ordering (or at least there was) which are precisely the variables for which the ordering is defined. But feel free to change this to something more sensible.

Reply:
Given that `MonomialOrdering`s seem to attached to polynomial rings, maybe we should make the properties of the ordering in relation to the whole ring. Will something else go wrong if we change it to that @joschmitt? I cannot estimate the consequences of changing the definition of these functions.

Reply:
The .vars member is still there for the same orderings, but I think old definition is adhoc and am in favor of changing it here.

Reply:
I don't think that anything relies on the "old definition", at least from my point of view it anyway only makes (mathematical) sense to ask `is_global` for an ordering that is defined on all of the monomials of the underlying ring (so that there is no ambiguity) and from what I can see these functions are used like that in OSCAR so far.
So as far as I'm concerned you can do whatever you like with this.

I know that this is pedantic now, but I looked up the definition of "global ordering" in the "Singular book" by Greuel and Pfister and it says "Let > be a monomial ordering on {x^\alpha | \alpha\in \mathbb N^n}. Then > is called a global ordering if x^\alpha > 1 for all \alpha \neq 0." (where x is the vector of variables x_1, ..., x_n.) So this definition only works for the monomials for which the ordering is defined. I am more and more thinking that e. g. `is_local(neglex([y, z]))` should in fact error, if the ring has more variables because we neither have x < 1 nor x > 1.
But no matter, I agree that the doc-string was ambiguously worded and I leave the definition to the experts.

Reply:
So the question is about what `o = neglex([y,z])` says about the comparison between `1` and `x` and if this matters for the definition of `is_local(p)`. The old orderings.jl and new orderings.jl both say they compare "equal", and it would introduce difficulties to say "no comparison possible". Let's see what @wdecker says.

Reply:
Thanks for your input @joschmitt. Much appreciated.

Reply:
@tthsqe12 The question whether `neglex([y, z])` defines a monomial ordering for which the predicates `is_local/is_global` should not error is orthogonal to the work you did here, right? Maybe it is best to discuss this later and merge this PR here.

Reply:
Sure, there are no uses is_local inside oscar.